### DISCUSSION TITLE
When running job in container, are CMD or ENTRYPOINT executed?

### DISCUSSION THREAD

**Discussion by petertiedemann** [community/community#26943] [Actions]
<p>When i run a job using “container: xxx/yyy:tag” will either CMD or ENTRYPOINT be executed? And will they have access to the “env:” defined for the job? E.g.</p>
<pre><code>env:
      NUGET_USER: ${{ secrets.NUGET_USER }}
      NUGET_PASS: ${{ secrets.NUGET_PASS }}
</code></pre>
<p>My scenario:</p>
<p>I run my jobs in a container to ensure that the test environment is completely well defined. The container contains an ENTRYPOINT that sets up a package source when executed based on secrets retrieved from environment variables ( i do not want those secrets in the container itself as it is public ). It seems that ENTRYPOINT is not executed. I figure maybe CMD might work, but i would like to understand if i can even access the environment variables.</p>
<p>Right now as a work around i have a startup.sh in my container that i execute in the job, but it adds one more line to every job.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/petertiedemann">@petertiedemann</a> ,</p>
<p>According to the syntax of container, there’s no CMD or entrypoint settings, only in the ‘<a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontaineroptions" rel="nofollow noopener">jobs.&lt;job_id&gt;.container.options</a>’ , it’s possible to overwrite the default ENTRYPOINT of the image with “–entrypoint”.</p>
<p>And during the initialization, container doesn’t have access to the “env” variables which defined in the job. </p>
<p>The env variables can be accessed in the job steps.</p>
<p>Thanks</p>

**Comment by petertiedemann** [comment]
<p>Thank you, so sounds like i will have to stick with running my startup.sh manually.</p>
